src/runtime/choice_point.rs

```diff
@@ -4,25 +4,26 @@ use serde::Deserialize;
 
 use crate::path::Path;
 
-#[derive(Debug, Deserialize)]
+#[derive(Debug, Deserialize, Default, PartialEq, Eq, Clone)]
+#[serde(from = "ChoicePointData")]
 pub struct ChoicePoint {
-    has_condition: bool,
-    has_start_content: bool,
-    has_choice_only_content: bool,
-    is_invisible_default: bool,
-    once_only: bool,
-    path_on_choice: Option<Path>,
+    pub has_condition: bool,
```

Comment:
This is beyond the scope of the pr, so I'm fine with it for now, but I'd like to discuss keeping fields private vs. making them public at some point